dnsaas/api/v2/views.py

```diff
@@ -313,3 +315,60 @@ class TsigKeysViewSet(FiltersMixin, ModelViewSet):
     queryset = TsigKey.objects.all()
     serializer_class = TsigKeysTemplateSerializer
     filter_fields = ('name', 'secret')
+
+
+class IPRecordView(APIView):
+    """Dedicated view for add/update/delete"""
+    permission_classes = (IsAdminUser,)
+
+    def _get_record(self, ip, hostname):
+        ip = int(ipaddress.ip_address(ip))
+        try:
+            record = Record.objects.get(number=ip, name=hostname)
+        except:
+            record = None
+        return record
+
+    def _add_or_update_record(self, data):
```

Comment:
TODO:
 - if ``hostname`` is empty in ``new`` dict then delete A record
 - if ``hostname`` is empty in ``old`` dict then create A record